default-placeholders/index.md

```diff
@@ -0,0 +1,269 @@
+which is used as the 5th accessor template parameter
+to indicate whether the accessor can be used as a placeholder.
+Only the `global_buffer` and `constant_buffer` access targets
+support placeholder accessors.
+
+The main reason for having placeholders is to store accessors into objects
+without having a queue at the point of object creation,
+and registering the access with a command group submission at a later time.
+One of Codeplay's proposals that didn't make it into SYCL 1.2.1
+was to allow placeholders to be default constructible as well,
+i.e. not having to even know the buffer at the point of construction.
+This extension is used in some SYCL ecosystem projects.
+
+Accessors are pointer-like objects,
+and placeholders try to fill that gap
+that prevents accessors from being even more pointer-like.
+A default constructed placeholder accessor is analogous to a null pointer.
+A placeholder that's bound to a buffer
+but hasn't been registered with a command group
+is more like a fancy pointer:
+the user doesn't own the data
+until the accessor is registered and used in a kernel,
+where is becomes more similar to a regular pointer.
+
+Having this type separation between full accessors and placeholders
+might be useful from a type safety perspective,
+but we believe it makes development more difficult.
+For example, [another one of our proposals](https://github.com/codeplaysoftware/standards-proposals/pull/100/files)
+introduces alias templates for different access targets
+and revises rules on how read-only accessors are handled,
+all in the name of reducing accessor verbosity.
+The placeholder template parameter makes that much more difficult,
+meaning we either need to introduce another parameter to the alias template,
+making it a lot less useful,
+or simply ignore the reduction in verbosity for placeholder accessors.
+
+## Changes
+
+### Deprecate `access::placeholder`
+
+Mark the `access::placeholder` enum class as deprecated,
+but keep it for backwards compatibility
+until it's eventually removed from a subsequent standard.
+
+```cpp
+namespace access {
+...
+
+enum class placeholder // deprecated
+{...};
+} // namespace access
+
+template <typename dataT,
+          int dimensions,
+          access::mode accessMode,
+          access::target accessTarget = access::target::global_buffer,
+          access::placeholder isPlaceholder = access::placeholder::false_t>
+class accessor;
+```
+
+### All accessors with a global or constant target can be placeholders
+
+SYCL 1.2.1 allows `access::placeholder::true_t`
+only when the access target is `global_buffer` or `constant_buffer`.
+We propose that the same placeholder semantics
+still apply to only these two targets,
+just that the semantics and API is available
+regardless of the `isPlaceholder` template parameter.
+
+### Accept all accessors in `handler::require`
+
+At the moment the member function `handler::require` only accepts
```

Comment:
Now that we allow an `accessor` to be null, we will need to define the behaviour when you pass a null `accessor` to `require`.

Reply:
I've clarified that `require` throws `cl::sycl::invalid_object_error` if a null accessor is passed. Although since calling `require` is allowed on both placeholders and non-placeholders, it might be worth allowing it for null accessors as well, though that might introduce new subtle errors.

Reply:
I think this is the right way to go, I think passing a null `accessor` to a kernel should be okay, as long as you don't dereference it, but calling `require` should fail as you're asking the runtime to create a data dependency on non-existent data. We could potentially allow it and have it be a no-op for generic programming sake, but if we have `is_null` I don't think that's necessary.